### ISSUE TITLE
feat(storage): pass file metadata through uploadFile to storage.put

### ISSUE THREAD

**Issue by JorgeMoralesLopez** [prescottprue/react-redux-firebase#720]
I'm doing login with facebook and when I enter the profile image in firebase storage I get a file with extension application / octet-stream, I need this file to have jpg, jpeg, png image format ... How can I put contentType?

`
 const firebaseLib = getFirebase()
 const imageName = uuid.v4(); 
 const path = `${user.uid}/user_images`
 const options = {
    name: imageName,
 };
const response = await urlToBlob(user.photoURL + '?type=large');
                
//upload the file to firebase storage
let uploadedFile = await firebaseLib.uploadFile(path, response, null, options)
// console.log('uploadedFile', uploadedFile);
// get url of image de firebase storage
let downloadURL = await uploadedFile.uploadTaskSnapshot.ref.getDownloadURL();
`

**Comment by prescottprue** [comment]
You can specify `contentType` as part of your call as shown [in the firebase storage docs](https://firebase.google.com/docs/storage/web/upload-files#add_file_metadata)

**Comment by JorgeMoralesLopez** [comment]
Thank you for your reply @prescottprue ... Yes, I've tried it several times but it does not work.
`const newMetadata = {
    contentType: 'image/jpeg'
 }
let uploadedFile = await firebaseLib.uploadFile(path, response, newMetadata, options);`

**Comment by prescottprue** [comment]
@JorgeMoralesLopez It actually appears that file metadata wasn't being passed along, so I started to work on that as a new feature for an upcoming version. The new syntax will look like so:

```js
const storagePath = 'avatars'
const dbPath = 'avatarFilesInfo'
const fileMetadata = { contentType: 'image/jpeg' }

firebase
  .uploadFile(storagePath, file, dbPath, { metadata: fileMetadata })
  .then(() => {
    console.log('File uploaded successfully')
  })
```

**Comment by JorgeMoralesLopez** [comment]
Thank you @prescottprue, I'm glad to hear that I was not on the wrong path because I searched the web and tried many ways.
If I upgrade to the new version, would I have to update other libraries like firebase, redux, react-redux, redux-firestore, thunk?

"firebase": "^5.4.1",
"react": "16.8.3",
"react-native": "https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz", (0.59 version)
"react-redux": "^5.0.7",
"react-redux-firebase": "^2.2.4",
"redux": "^4.0.0",
"redux-firestore": "^0.6.0",
"redux-thunk": "^2.3.0",

**Comment by prescottprue** [comment]
Fix included in [v2.3.0 release](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.3.0) including a docs update with an example. Reach out if it doesn't work as expected. Thanks for reporting!

**Comment by fuhrmannb** [comment]
I comment this issue because the feature was not released on `next` version.
Is it possible to add the feature also on the beta release?
